src/win/getaddrinfo.c

```diff
@@ -268,51 +279,52 @@ int uv_getaddrinfo(uv_loop_t* loop,
       return rc;
     nodesize = strlen(hostname_ascii) + 1;
     node = hostname_ascii;
+    off += nodesize * sizeof(WCHAR);
   }
 
   if (service != NULL) {
     rc = uv_wtf8_length_as_utf16(service);
     if (rc < 0)
        return rc;
     servicesize = rc;
+    off = align_offset(off, sizeof(WCHAR));
+    serviceoff = off;
+    off += servicesize * sizeof(WCHAR);
   }
+
   if (hints != NULL) {
-    hintssize = ALIGNED_SIZE(sizeof(struct addrinfoW));
+    off = align_offset(off, sizeof(void *));
+    hintoff = off;
+    hintssize = sizeof(struct addrinfoW);
+    off += hintssize;
   }
 
   /* allocate memory for inputs, and partition it as needed */
-  alloc_ptr = uv__malloc(ALIGNED_SIZE(nodesize * sizeof(WCHAR)) +
-                         ALIGNED_SIZE(servicesize * sizeof(WCHAR)) +
-                         hintssize);
-  if (!alloc_ptr)
+  req->alloc = uv__malloc(off);
+  if (!req->alloc)
     return UV_ENOMEM;
 
-  /* save alloc_ptr now so we can free if error */
-  req->alloc = (void*) alloc_ptr;
-
   /* Convert node string to UTF16 into allocated memory and save pointer in the
    * request. The node here has been converted to ascii. */
   if (node != NULL) {
-    req->node = (WCHAR*) alloc_ptr;
-    uv_wtf8_to_utf16(node, (WCHAR*) alloc_ptr, nodesize);
-    alloc_ptr += ALIGNED_SIZE(nodesize * sizeof(WCHAR));
+    req->node = (WCHAR *)req->alloc;
+    uv_wtf8_to_utf16(node, req->node, nodesize);
   } else {
     req->node = NULL;
   }
 
   /* Convert service string to UTF16 into allocated memory and save pointer in
    * the req. */
   if (service != NULL) {
-    req->service = (WCHAR*) alloc_ptr;
-    uv_wtf8_to_utf16(service, (WCHAR*) alloc_ptr, servicesize);
-    alloc_ptr += ALIGNED_SIZE(servicesize * sizeof(WCHAR));
+    req->service = (WCHAR *)((char *)req->alloc + serviceoff);
```

Comment:
style nit: for consistency (here and elsewhere), try to keep the the `*` leaning left with a space after the cast (sorry we don't have any clang-format rules setup to make this easy for you)
```suggestion
    req->service = (WCHAR*) ((char*) req->alloc + serviceoff);
```